Assets/LeapMotion/Core/Scripts/Streams/StreamConnector.cs

```diff
@@ -0,0 +1,197 @@
+    /// </summary>
+    public abstract System.Type streamDataType { get; }
+
+    /// <summary>
+    /// Gets the bounded generic type for the input IStream, whose generic type argument
+    /// MUST match streamDataType.
+    /// </summary>
+    public abstract System.Type streamType { get; }
+
+    /// <summary>
+    /// Gets the bounded generic type for the connector's output type IStreamReceiver,
+    /// whose generic type argument MUST match streamDataType.
+    /// </summary>
+    public abstract System.Type streamReceiverType { get; }
+  }
+
+  [ExecuteInEditMode]
+  public abstract class StreamConnector<StreamType> : StreamConnector,
+                                                      IRuntimeGizmoComponent,
+                                                      ISerializationCallbackReceiver {
+
+    [SerializeField]
+    // Sadly, this doesn't compile (name isn't constant), so we'll have to
+    // use a Custom Editor instead
+    //[ImplementsInterface(typeof(IStream<StreamType>).AssemblyQualifiedName)]
+    protected MonoBehaviour _stream;
+    public IStream<StreamType> stream {
+      get { return _stream as IStream<StreamType>; }
+    }
+
+    [SerializeField]
+    // Sadly, this doesn't compile (name isn't constant), so we'll have to
+    // use a Custom Editor instead
+    //[ImplementsInterface(typeof(IStreamReceiver<StreamReceiverType>)
+    //                       .AssemblyQualifiedName)]
+    protected MonoBehaviour _receiver;
+    public IStreamReceiver<StreamType> receiver {
+      get { return _receiver as IStreamReceiver<StreamType>; }
+    }
+
+    public void OnBeforeSerialize() {
+
+    }
+
+    public void OnAfterDeserialize() {
+      // GO GO GO!
+      initCallbacks();
+    }
+
+    #region StreamConnector
+
+    /// <summary>
+    /// Gets the Type corresponding to the IStream and IStreamReceiver type arguments
+    /// that this Connector connects.
+    /// </summary>
+    public override System.Type streamDataType {
+      get { return typeof(StreamType); }
+    }
+
+    /// <summary>
+    /// Gets the bounded generic type for the input IStream, whose generic type argument
+    /// matches streamDataType.
+    /// </summary>
+    public override System.Type streamType {
+      get { return typeof(IStream<StreamType>); }
+    }
+
+    /// <summary>
+    /// Gets the bounded generic type for the connector's output type IStreamReceiver,
+    /// whose generic type argument matches streamDataType.
+    /// </summary>
+    public override System.Type streamReceiverType {
+      get { return typeof(IStreamReceiver<StreamType>); }
+    }
+
+    #endregion
+
+    public bool drawWire = true;
+
+    [OnEditorChange("ResetConnection")]
+    public bool debugWire = false;
+
+    protected void ResetConnection() {
+      OnDisable();
+      OnEnable();
+    }
+
+    private void Awake() {
+      initCallbacks();
+    }
+
+    private void OnEnable() {
+      initCallbacks();
```

Comment:
Making a note here, if the wire is disabled or enabled at the right time, it can cause the receiver to receive callbacks in an illegal order (two opens in a row without a close for example)